src/text/countvectorizer.ts

```diff
@@ -0,0 +1,76 @@
+import Counter from './counter';
+
+interface StringIntegerObject {
+  [property: string] : number
+}
+export default class CountVectorizer {
+  public wordOrder: StringIntegerObject = {};
+  public uniqueLength: number;
+  /**
+   * Convert text to vector base on their number of count
+   * and the order to which they occur alphabetically.
+   * @param textArray
+   */
+  constructor(textArray: string[]) {
+
+    const tokenArray: string[] = [];
+    textArray.forEach((value) => {
+      value.split(" ").forEach((text) => {
+        if (text != '') {
+          tokenArray.push(text.toLocaleLowerCase().replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g, ''));
+        }
+      });
+    });
+
+    const uniqueWord = Array.from(new Set(tokenArray));
+    const sortedUniqueWord: string[] = this.sort(uniqueWord);
+
+    // store array element as object element
+    // to enable quick search of element
+    // when transforming input arrays.
+    sortedUniqueWord.forEach((text, index) => {
+      if (text != '') {
+        this.wordOrder[text] = index;
+      }
+    });
+    this.uniqueLength = sortedUniqueWord.length;
+
+  }
+
+  private sort(textArray: string[]): string[] {
+    const collator = new Intl.Collator(undefined, { numeric:true, sensitivity: 'base' });
+
+    return textArray.sort(collator.compare);
+  }
+  /**
+   * Transform input text to vector.
+   * @param textArray
+   * @returns number[][] array of number (vectors)
+   */
+  public transform(textArray: string[]): number[][] {
+
+    const counterVectorizer: number[][] = textArray.map((value) => {
+      const innerArray: number[] = Array.from(new Float64Array(this.uniqueLength));
+      const cleanTextArray: string[] = [];
+      value.split(" ").forEach((text) => {
```

Comment:
Ditto, we need to unify the format to use quotes :)

Reply:
ok